thunderbird/processes/wps_update_metadata.py

```diff
@@ -92,6 +95,37 @@ def copy_and_get_filepath(self, request):
         shutil.copyfile(original, copy)
         return copy
 
+    def updates_instruction_generator(self, request):
+        """
+        This function generates a dictionary containing the updates instruction information.
+        The dictionary will be constructed using the content of the input updates_file/updates_string.
+        It is desired that only one of the options is provided by the user, but if both are given, 
+        updates_file input will be used to generate whiile updates_string input will be neglected.
+
+        There are odd behaviours of the ComplexInput updates_file; the content of the yaml file is 
+        directly accessible using the .data attribute when run in localhost while it is accessible 
+        as '.url' attribute when run in docker containers. When running pytest, the .data attribute
+        holds the path to the origial file path.
+        """
+        if "updates_file" in request.inputs.keys():
+            try:
+                updates = request.inputs["updates_file"][
+                    0
+                ].data  # For running in localhost
+            except exceptions.NoApplicableCode:
+                updates = request.inputs["updates_file"][
+                    0
+                ].url  # For running in Docker containers
+        elif "updates_string" in request.inputs.keys():
+            updates = request.inputs["updates_string"][0].data
+
+        # Convert yaml content to a dictionary
+        if os.path.isfile(updates):  # For running pytest
+            with open(updates) as ud:
+                return yaml.safe_load(ud)
```

Comment:
Since this is for `pytest`, is there a way we could deal with this in the tests rather than the source code? What exactly is this purpose of this blurb?

Reply:
As I told you before, running a `pywps` process creates temporary files named `input.extension` in `/tmp` directory and copies the content of `ComplexInput` to the file. However, it is a myth that when running the processes in pytest does not copy the content, but instead it stores the path to the original file in the `input.extension` file. It is very odd. There's gotta be an internal issue in how pytest runs the processes, but I could not identify the problem yet. If the issue is solved, that `if` block can be removed.